src/internal/storage/chunk/storage.go

```diff
@@ -18,50 +19,59 @@ const (
 // Storage is the abstraction that manages chunk storage.
 type Storage struct {
 	objClient obj.Client
+	store     kv.Store
+	memCache  kv.GetPut
 	tracker   track.Tracker
 	mdstore   MetadataStore
 
-	defaultChunkTTL time.Duration
+	createOpts CreateOptions
 }
 
 // NewStorage creates a new Storage.
-func NewStorage(objClient obj.Client, mdstore MetadataStore, tracker track.Tracker, opts ...StorageOption) *Storage {
+func NewStorage(objC obj.Client, memCache kv.GetPut, mdstore MetadataStore, tracker track.Tracker, opts ...StorageOption) *Storage {
 	s := &Storage{
-		objClient:       objClient,
-		mdstore:         mdstore,
-		defaultChunkTTL: defaultChunkTTL,
-		tracker:         tracker,
+		objClient: objC,
+		memCache:  memCache,
+		mdstore:   mdstore,
+		tracker:   tracker,
+		createOpts: CreateOptions{
+			Compression: CompressionAlgo_GZIP_BEST_SPEED,
+		},
 	}
 	for _, opt := range opts {
 		opt(s)
 	}
+	s.store = kv.NewFromObjectClient(s.objClient)
+	s.objClient = nil
```

Comment:
Why do we still have this field if we are just going to nil it out?

Reply:
So that the options can configure it.  I set it to nil so anything still referencing it would break.

Reply:
Okay, not too sure how I feel about this, but I guess it is fine for now.